frontend/ui/widget/toggleswitch.lua

```diff
@@ -65,27 +65,38 @@ function ToggleSwitch:init()
     local item_padding = Size.padding.default -- only used to check if text truncate needed
     local item_border_size = Size.border.thin
     local frame_inner_width = self.width - 2*self.toggle_frame.padding - 2* self.toggle_frame.bordersize
-    local item_width = math.ceil(frame_inner_width / self.n_pos - 2*item_border_size)
-    local item_height = self.height / self.row_count
+    -- We'll need to adjust items width and distribute the accumulated fractional part to some
+    -- of them for proper visual alignment
+    local item_width_real = frame_inner_width / self.n_pos - 2*item_border_size
+    local item_width = math.floor(item_width_real)
+    local item_width_adjust = item_width_real - item_width
     -- Note: the height provided by ConfigDialog might be smaller than needed,
     -- it gets too thin if we account for padding & border
-    local center_dimen = Geom:new{
-        w = item_width,
-        h = item_height,
-    }
+    local item_height = self.height / self.row_count
+    local item_width_to_add = 0
     for i = 1, #self.toggle do
+        local real_item_width = item_width
+        item_width_to_add = item_width_to_add + item_width_adjust
+        if item_width_to_add >= 1 then
+            -- One pixel narrower to better align the entire widget
```

Comment:
Looks good to me.
Please just update that comment, narrower => wider (now that you changed the logic :)